org.eclipse.buildship.ui/src/main/java/org/eclipse/buildship/ui/internal/preferences/GradleCompositeImportOptionsPreferencePage.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright (c) 2017 the original author or authors.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+
+package org.eclipse.buildship.ui.internal.preferences;
+
+import java.io.File;
+
+import org.eclipse.buildship.core.internal.CorePlugin;
+import org.eclipse.buildship.core.internal.configuration.BuildConfiguration;
+import org.eclipse.buildship.core.internal.configuration.ConfigurationManager;
+import org.eclipse.buildship.core.internal.util.binding.Validator;
+import org.eclipse.buildship.core.internal.util.binding.Validators;
+import org.eclipse.buildship.ui.internal.util.gradle.GradleDistributionViewModel;
+import org.eclipse.buildship.ui.internal.util.widget.AdvancedOptionsGroup;
+import org.eclipse.buildship.ui.internal.util.widget.GradleProjectSettingsComposite;
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.runtime.Platform;
+import org.eclipse.swt.events.SelectionEvent;
+import org.eclipse.swt.events.SelectionListener;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Control;
+import org.eclipse.ui.dialogs.PreferencesUtil;
+import org.eclipse.ui.dialogs.PropertyPage;
+
+/**
+ * Preference page for Gradle projects.
+ *
+ * @author Donat Csikos
+ */
+public final class GradleCompositeImportOptionsPreferencePage extends PropertyPage {
+
+    public static final String PAGE_ID = "org.eclipse.buildship.ui.compositeImportOptionsProperties";
+
+    private GradleProjectSettingsComposite gradleProjectSettingsComposite;
+
+    private final Validator<GradleDistributionViewModel> distributionValidator;
```

Comment:
⭕️  Do we really need validators at this point? Judging by the code only I should be able to get validation for the widgets (for invalid Gradle distributions for instance), but it's not triggering. 

I'm fine leaving the validation in the code though, I just want to understand your reasoning.

Reply:
I just kept the page exactly how it was because I assumed, that the implementation for the import preferences would stay the same.